pandas/core/categorical.py

```diff
@@ -325,12 +320,13 @@ def __init__(self, values, categories=None, ordered=False, fastpath=False):
 
             # make sure that we always have the same type here, no matter what
             # we get passed in
-            categories = self._validate_categories(categories)
-            codes = _get_codes_for_values(values, categories)
+            dtype = CategoricalDtype(categories, ordered)
+            codes = _get_codes_for_values(values, dtype.categories)
 
             # TODO: check for old style usage. These warnings should be removes
```

Comment:
maybe in followup (do we have an issue for this?). we should remove this and instead raise I think.

Reply:
Didn't see one, opened https://github.com/pandas-dev/pandas/issues/17485